packages/fluentui/react-northstar/src/utils/getReactFiberFromNode.ts

```diff
@@ -0,0 +1,259 @@
+  // // before its child fibers are created.
+  // mode: TypeOfMode
+  //
+  // // Effect
+  // effectTag: SideEffectTag
+
+  // Singly linked list fast path to the next fiber with side-effects.
+  nextEffect: Fiber | null;
+
+  // The first and last fiber with side-effect within this subtree. This allows
+  // us to reuse a slice of the linked list when we reuse the work done within
+  // this fiber.
+  firstEffect: Fiber | null;
+  lastEffect: Fiber | null;
+
+  // Represents a time in the future by which this work should be completed.
+  // Does not include work found in its subtree.
+  expirationTime: ExpirationTime;
+
+  // This is used to quickly determine if a subtree has no pending changes.
+  childExpirationTime: ExpirationTime;
+
+  // This is a pooled version of a Fiber. Every fiber that gets updated will
+  // eventually have a pair. There are cases when we can clean up pairs to save
+  // memory if we need to.
+  alternate: Fiber | null;
+
+  // Time spent rendering this Fiber and its descendants for the current update.
+  // This tells us how well the tree makes use of sCU for memoization.
+  // It is reset to 0 each time we render and only updated when we don't bailout.
+  // This field is only set when the enableProfilerTimer flag is enabled.
+  actualDuration?: number;
+
+  // If the Fiber is currently active in the "render" phase,
+  // This marks the time at which the work began.
+  // This field is only set when the enableProfilerTimer flag is enabled.
+  actualStartTime?: number;
+
+  // Duration of the most recent render time for this Fiber.
+  // This value is not updated when we bailout for memoization purposes.
+  // This field is only set when the enableProfilerTimer flag is enabled.
+  selfBaseDuration?: number;
+
+  // Sum of base times for all descendants of this Fiber.
+  // This value bubbles up during the "complete" phase.
+  // This field is only set when the enableProfilerTimer flag is enabled.
+  treeBaseDuration?: number;
+
+  // Conceptual aliases
+  // workInProgress : Fiber ->  alternate The alternate used for reuse happens
+  // to be the same as work in progress.
+  // __DEV__ only
+  _debugID?: number;
+  _debugSource?: Source | null;
+  _debugOwner?: Fiber | null;
+  _debugIsCurrentlyTiming?: boolean;
+  _debugNeedsRemount?: boolean;
+
+  // Used to verify that the order of hooks does not change between renders.
+  _debugHookTypes?: HookType[] | null;
+};
+
+export function getReactFiberFromNode(elm: Node | undefined): Fiber | null {
+  if (!elm) {
+    return null;
+  }
+
+  for (const k in elm) {
+    // React 16 uses "__reactInternalInstance$" prefix
+    // React 17 uses "__reactFiber$" prefix
+    // https://github.com/facebook/react/pull/18377
+    if (k.indexOf('__reactInternalInstance$') === 0 || k.indexOf('__reactFiber$') === 0) {
+      return elm[k];
+    }
+  }
+
+  throw new Error('getReactFiber(): Failed to find a React Fiber on a node');
```

Comment:
nit: consider having a test case for the throw

Reply:
Added 👍